modules/message.sh

```diff
@@ -275,20 +285,53 @@ function message {
           elif check_apitoken slack; then
             get_apitoken slack
           else
-            echo "To get an authorized access token"
+            echo "To get an authorized access token:"
             echo ""
-            echo "Navigate to https://api.slack.com/apps and create an APP. Provide a name for the APP and select the \"Development Slack Workspace (eg : Open Learning Exchange)\" from the drop down list"
-            echo "Go to \"OAuth & Permission\" under \"features \" and select the scope under \"User Token Scopes\" and add \"chat:write\", \"channels:write\", \"channel:read\", \"channel:history\", \"groups:write\", \"group:read\", \"mpim:write\", \"im:write\", \"usergroups.read\" and \"users.read\" for the APP from the drop down list"
-            echo "Then install APP to the workspace and click the allow button to give permissions in the redirected link and then you will get the \"OAuth access token\""
-            echo "Run $BASENAME message slack apitoken <oauth access token>"
+            echo "1. Go to https://api.slack.com/apps"
+            echo "2. Click \"Create New App\""
+            echo "3. Type a name in \"App Name\""
+            echo "4. Select the \"Development Slack Workspace\" (eg : Open Learning Exchange)\ from the drop-down"
+            echo "5. Click \"OAuth & Permission\" under \"features \""
+            echo "6. Under \"User Token Scopes\", click \"Add an OAuth Scope\""
+            echo "7. Add the following permissions:"
+            echo "  \"chat:write\""
+            echo "  \"channels:write\""
+            echo "  \"channels:read\""
+            echo "  \"channels:history\""
+            echo "  \"groups:write\""
+            echo "  \"groups:read\""
+            echo "  \"im:write\""
+            echo "  \"mpim:write\""
+            echo "  \"usergroups:read\""
+            echo "  \"users:read\""
+            echo "8. Under \"OAuth Tokens & Redirect URLs\", click \"Install to Workspace\""
+            echo "9. Click \"Allow\". This will give permissions and display the User OAuth Token"
+            echo "10. Run $BASENAME message slack apitoken <User OAuth Token>"
           fi
           ;;
         channels)
           if check_apitoken slack; then
-            channel_names=$(get_channel_slack)
-            echo "Channels Names:"
-            echo
-            echo "$channel_names"
+            if channel_names=$(get_channel_slack); then
+              echo "Channels Names:"
+              echo
+              echo "$channel_names"
+            else
+              needed=$(curl -s -F token=$access_token -F types=public_channel,private_channel https://slack.com/api/users.conversations | jq '."needed"')
```

Comment:
So I ran the code and the jq error is now gone but it did not print out any missing scope. I feel like the code for the missing scope will never reach since the curl on line 319 will always return null if **any** of the required scope returns true.

The reason behind this is that `channels:read  groups:read  im:read  mpim:read` can run the `https://slack.com/api/users.conversations` method. The documentation page https://api.slack.com/methods/users.conversations.

`jq .needed` will not filter since the needed parameter will never return from the request, only the ok request. Hence never reaching the missing scope conditionals.

Reply:
What permissions are you using? I'd like to duplicate this on my end for debugging.

Reply:
\"chat:write\""
\"channels:write
\"channels:read\
\"channels:histo
\"groups:write\"
\"im:write\""
\"mpim:write\""
\"usergroups:read
\"users:read\""

API token: xoxp-1269890176869-1840115517223-1852008762647-870b3cb5aff2dd0f87ffcba1c3c059f1

-------

\"chat:write\""

API token: xoxp-1269890176869-1840115517223-1882870085380-6fdb99eb13eafc2a769ea19497499842

-------

\"chat:write\""
\"channels:write
\"channels:history
\"groups:write\"
\"groups:read\"
\"im:write\""
\"mpim:write\""
\"usergroups:read


API token: xoxp-1269890176869-1840115517223-1869876839558-ce316ff3066cda81feb05577b02f1aad